src/arg.c

```diff
@@ -110,6 +110,162 @@ static void dcc_note_compiled(const char *input_file, const char *output_file)
+        return 1;
+    }
+    int outRead = outfds[0];
+    int outWrite = outfds[1];
+    int inRead = infds[0];
+    int inWrite = infds[1];
+
+    switch (fork()) {
+    case -1:
+        fprintf(stderr, "Failed to fork: %s (%d)", strerror(errno), errno);
+        return 1;
+    case 0: { // Child
+        close(outWrite);
+        dup2(outRead, STDIN_FILENO);
+        close(outRead);
+        close(inRead);
+        dup2(inWrite, STDERR_FILENO);
+        close(inWrite);
+        execlp(compiler, compiler, "-v", "-E", "-x", "c",
+               found_arch_native ? "-march=native" : "-DDUMMY",
+               found_tune_native ? "-mtune=native" : "-DDUMMY",
+               found_cpu_native ? "-mcpu=native" : "-DDUMMY",
+               "-", NULL);
+        return 0;
+    }
+    }
+
+    close(outWrite);
+    close(outRead);
+    close(inWrite);
+
+    FILE *in = fdopen(inRead, "r");
+    while(fgets(buff, sizeof(buff), in) != NULL)
+    {
+        char* cc1 = strstr(buff, "cc1");
+        if (cc1)
+        {
+            char* args = strstr(cc1, " - ");
+            if (args)
+            {
+                args += 3;
+                char* end = strstr(args, " -v ");
+                if (!end) end = strstr(args, "\n");
+                if (end) *end = 0;
+                int n_arguments = 1;
+                for (const char* ptr = args;*ptr;ptr++) {
+                    if (*ptr == ' ') n_arguments++;
+                }
+                pclose(in);
+                close(inRead);
+
+                if (is_clang) n_arguments *= 2; /* must prepend all clang cc1 options with -Xclang */
+
+                char **b;
+                b = malloc((l+1+extra_args+n_arguments-1) * (sizeof argv[0]));
+                if (b == NULL) {
+                    rs_log_error("failed to allocate copy of argv");
+                    return EXIT_OUT_OF_MEMORY;
+                }
+                j = 0;
+                int args_replaced = 0;
+                for (i = 0; i < l; i++) {
+                    if (!strcmp(argv[i], "-march=native") || !strcmp(argv[i], "-mtune=native") ||
+                        !strcmp(argv[i], "-mcpu=native")) {
+                        if (args_replaced == 0) {
+                            /* insert only once, at the position of the first native argument */
+                            *ignore_range_min = j;
+                            const char* ptr_insert = args;
+                            int clang_force_next = 0;
+                            for (char* ptr = args;*ptr;ptr++) {
+                                if (*ptr == ' ' || *ptr == 0) {
+                                    if (*ptr == ' ') *(ptr++) = 0;
+                                    const char* insert = ptr_insert;
+                                    ptr_insert = ptr;
+                                    if (is_clang) {
+                                        if (strncmp(insert, "-target", strlen("-target")) == 0) {
```

Comment:
How can this possibly work when `-target` is set? Unless you know that the `-target` is the native target, there is nothing that can be done with `=native`.

Reply:
This is to remove `-target` if `-march=native`, `-mtune=native` or `-mcpu=native` I guess. I assumed it was because -target is implicitly set by -mcpu/-march/-mtune, but please correct me if it's wrong (the man-page for clang is ... not very verbose on what `-target` does).

Reply:
If you remove `-target` then you break cross-compiling.

Reply:
I would not bother for `-target` too much for code that is meant for march=native. In that case we cannot cross-compile anyway